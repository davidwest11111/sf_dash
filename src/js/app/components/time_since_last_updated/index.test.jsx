import React from 'react';
import { shallow } from 'enzyme';
import shallowToJson from 'enzyme-to-json';
import moment from 'moment';
import 'moment-precise-range-plugin';

import TimeSinceLastUpdated, { INTERVAL_MS } from './index';


describe('outputs the expected tree when', () => {
  let wrapper;

  test('lastUpdated not passed', () => {
    wrapper = shallow((
      <TimeSinceLastUpdated />
    ));
  });

  test('lastUpdated is now', () => {
    wrapper = shallow((
      <TimeSinceLastUpdated lastUpdated={moment()} />
    ));
  });

  test('the update is not overdue', () => {
    wrapper = shallow((
      <TimeSinceLastUpdated lastUpdated={moment().subtract(1, 'seconds')} />
    ));
  });

  test('the update is overdue', () => {
    wrapper = shallow((
      <TimeSinceLastUpdated
        lastUpdated={moment().subtract(5, 'seconds')}
        msUntilWarning={3000}
      />
    ));
  });

  test('is loading', () => {
    wrapper = shallow((
      <TimeSinceLastUpdated lastUpdated={moment()} loading />
    ));
  });

  afterEach(() => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});

describe('instance methods', () => {
  let wrapper;
  let instance;
  let setIntervalSpy;
  let clearIntervalSpy;

  beforeEach(() => {
    setIntervalSpy = jest.spyOn(window, 'setInterval').mockReturnValue(11);
    clearIntervalSpy = jest.spyOn(window, 'clearInterval');
    wrapper = shallow((
      <TimeSinceLastUpdated />
    ));
    instance = wrapper.instance();
  });

  test('componentDidMount sets an interval', () => {
    expect(setIntervalSpy).toHaveBeenCalledWith(expect.any(Function), INTERVAL_MS);
    expect(instance.interval).toEqual(11);
  });

  test('componentWillUnmount clears an interval', () => {
    wrapper.unmount();

    expect(clearIntervalSpy).toHaveBeenCalledWith(11);
  });
});
