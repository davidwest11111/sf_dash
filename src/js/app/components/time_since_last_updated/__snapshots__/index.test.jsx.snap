// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`outputs the expected tree when is loading 1`] = `
<div
  className="time-since-last-updated"
>
  <span
    className=""
  >
    Just now.
  </span>
  <Loader
    className="my-1 ml-auto"
    loading={true}
    small={true}
  />
</div>
`;

exports[`outputs the expected tree when lastUpdated is now 1`] = `
<div
  className="time-since-last-updated"
>
  <span
    className=""
  >
    Just now.
  </span>
  <Loader
    className="my-1 ml-auto"
    loading={false}
    small={true}
  />
</div>
`;

exports[`outputs the expected tree when lastUpdated not passed 1`] = `
<div
  className="time-since-last-updated"
>
  <span
    className=""
  />
  <Loader
    className="my-1 ml-auto"
    loading={false}
    small={true}
  />
</div>
`;

exports[`outputs the expected tree when the update is not overdue 1`] = `
<div
  className="time-since-last-updated"
>
  <span
    className=""
  >
    1 second ago.
  </span>
  <Loader
    className="my-1 ml-auto"
    loading={false}
    small={true}
  />
</div>
`;

exports[`outputs the expected tree when the update is overdue 1`] = `
<div
  className="time-since-last-updated"
>
  <span
    className="overdue"
  >
    5 seconds ago.
  </span>
  <Loader
    className="my-1 ml-auto"
    loading={false}
    small={true}
  />
</div>
`;
